python/cuml/test/test_dbscan.py

```diff
@@ -60,22 +65,49 @@ def test_dbscan_predict_numpy(datatype):
                  dtype=datatype)
 
     print("Calling fit_predict")
-    cudbscan = cuDBSCAN(eps=3, min_samples=2)
+    handle, stream = get_handle(use_handle)
+    cudbscan = cuDBSCAN(handle=handle, eps=3, min_samples=2)
     cu_labels = cudbscan.fit_predict(gdf)
     skdbscan = skDBSCAN(eps=3, min_samples=2)
     sk_labels = skdbscan.fit_predict(X)
     print(X.shape[0])
     for i in range(X.shape[0]):
         assert cu_labels[i] == sk_labels[i]
+    cudbscan.sync()
+
+
+def test_dbscan_predict_multiple_streams():
+    datatype = np.float32
+    gdf = cudf.DataFrame()
+    gdf['0'] = np.asarray([1, 2, 2, 8, 8, 25], dtype=datatype)
+    gdf['1'] = np.asarray([2, 2, 3, 7, 8, 80], dtype=datatype)
+
+    X = np.array([[1, 2], [2, 2], [2, 3], [8, 7], [8, 8], [25, 80]],
+                 dtype=datatype)
+
+    skdbscan = skDBSCAN(eps=3, min_samples=2)
+    sk_labels = skdbscan.fit_predict(X)
+
+    handle1, stream1 = get_handle(True)
+    handle2, stream2 = get_handle(True)
+    cudbscan1 = cuDBSCAN(handle=handle1, eps=3, min_samples=2)
+    cudbscan2 = cuDBSCAN(handle=handle2, eps=3, min_samples=2)
+    cu_labels1 = cudbscan1.fit_predict(gdf)
+    cu_labels2 = cudbscan2.fit_predict(gdf)
+    for i in range(X.shape[0]):
+        assert cu_labels1[i] == sk_labels[i]
+        assert cu_labels2[i] == sk_labels[i]
+    cudbscan1.sync()
+    cudbscan2.sync()
```

Comment:
The syncs in line 100 and 101 are needed before the loop in line 97 starts, no?

Reply:
Done